simulation.py

```diff
@@ -733,6 +767,37 @@ def compute_metrics(self, time_step, **kwargs):
                     self.metrics[f"global_clustering_{i}"].append(self.dynamics.network.global_clustering(grp=grp))
                     self.metrics[f"gini_coeff_{i}"].append(self.dynamics.network.gini_coeff(grp=grp))
 
+class AB_Experiment(Experiment):
+    def __init__(self, grp_info: 'list[dict]', treatment_probability=0, treatment_size=0, **kwargs):
+        kwargs['treatment_probability'] = treatment_probability
+        super().__init__(grp_info, **kwargs)
+        # out of the initialized nodes: sample treatment_probability * total nodes to be treatment
+        # this need to modify the existing network at time step 0
+        # thoughout the dynamics; whenever a new node is added, toss a coin with probability treatment_probability
+        # and assign the node to the treatment group
+        # two options here: fixed number of treatment nodes or fixed proportion of treatment nodes
+        if treatment_probability:
+            assert treatment_probability <= 1
+            flips = np.random.random_sample(size=self.dynamics.network.nodes) < treatment_probability
+            self.treatment_nodes = [i for i, f in zip(range(self.dynamics.network.nodes), flips) if f]
+        elif treatment_size:
+            assert treatment_size <= self.dynamics.network.nodes
+            self.treatment_nodes = random.sample(list(self.dynamics.network.G), treatment_size)
+        self.dynamics.network.assign_treatment(self.treatment_nodes)
```

Comment:
given that we only have a minimal change in the constructor; maybe we don't even need a separate AB_Experiement class; let's just put these changes in the `Experiment` class

Reply:
Sure! done